src/item_location.cpp

```diff
@@ -31,7 +33,12 @@ class item_location::impl
         virtual void remove_item() = 0;
 
     protected:
-        item *what = nullptr;
+        impl( item_uid &&uid ) : uid( std::move( uid ) ) {}
```

Comment:
That's a problem with implicit generation of uids, not general copy semantics.  If you have to std::move these things around to keep them valid you've made them very fragile.